qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -77,15 +77,22 @@ def __init__(
                 dense return array independent of the other settings.
             interpret: A callable that maps the measured integer to another unsigned integer or
                 tuple of unsigned integers. These are used as new indices for the (potentially
-                sparse) output array. If this is used, the output shape of the output needs to be
-                given as a separate argument.
-            output_shape: The output shape of the custom interpretation. The output shape is
-                automatically determined in case of sampling==True.
+                sparse) output array. If this is used, and sampling==False, the output shape of
+                the output needs to be given as a separate argument.
+            output_shape: The output shape of the custom interpretation, only used in the case
+                where an interpret function is provided and sampling==False. Note that in the
+                remaining cases, the output shape is automatically inferred by: `2^num_qubits` if
+                sampling==False and interpret==None, `(num_samples,1)` if sampling==True and
+                interpret==None, and `(num_samples,interpret_shape)` if sampling==True and an
+                interpret function is provided.
             gradient: The gradient converter to be used for the probability gradients.
-            quantum_instance: The quantum instance to evaluate the circuits.
+            quantum_instance: The quantum instance to evaluate the circuits. Note that
+                if sampling==True, 'statevector_simulator' is not a valid backend for the
```

Comment:
Can we avoid the use of a backend name here ie. `statevector_simulator` which is no longer supported for Aer anyway; rather simply say something like: that if sampling is True a statevector simulator is not a valid backend and will cause an error to be raised. Where statevector simulator is more just a description rather than a name - this is more how the error message itself looks.

Reply:
That makes sense, I did take it from the error message... Will do, thanks!